tests/unit/test_event.py

```diff
@@ -0,0 +1,158 @@
+    event_replicator.config = config
+    event_replicator.client = event_replicator.config.cognite.get_cognite_client("test_event_replicator")
+    event_replicator.state_store = LocalStateStore(
+        event_replicator.config.extractor.state_store.local.path
+    )
+
+    return event_replicator
+
+
+@pytest.fixture
+def event_data():
+    return [
+        {
+            'external_id': 'Test_Early_Notification_1710502959241', 
+            'data_set_id': 3351080165451477, 
+            'start_time': 1710502959241, 
+            'end_time': 1710503079241, 
+            'type': 'Notification', 
+            'subtype': 'HE3', 
+            'description': 'Failed Instrumentation Fire extinguisher test', 
+            'metadata': {'description': 'Failed Instrumentation Fire extinguisher test', 'notification': '22081904', 'work_order_number': '22082901', 'notification_type': 'HE3', 'work_center': 'SUPV'}, 
+            'asset_ids': [5534771458240925], 
+            'id': 3631277804748961, 
+            'last_updated_time': 1710505316426, 
+            'created_time': EARLY_CREATED_TIME
+        },
+        {
+            'external_id': 'Test_Late_Notification_1710506390376', 
+            'data_set_id': 3351080165451477, 
+            'start_time': 1710506390376, 
+            'end_time': 1710506510376, 
+            'type': 'Notification', 
+            'subtype': 'HE3', 
+            'description': 'Failed Instrumentation Fire extinguisher test', 
+            'metadata': {'description': 'Failed Instrumentation Fire extinguisher test', 'notification': '22081904', 'notification_type': 'HE3', 'work_center': 'SUPV', 'work_order_number': '22082901'}, 
+            'asset_ids': [5534771458240925], 
+            'id': 5875589750425323, 
+            'last_updated_time': 1710506651232, 
+            'created_time': LATE_CREATED_TIME
+        }
+    ]
+
+@pytest.fixture
+def mock_get_events(mocker, event_data):
+    return_value = iter([EventList([Event(**data) for data in event_data])])
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.get_events", return_value=return_value)
+
+
+@pytest.fixture
+def mock_get_late_events(mocker, event_data):
+    return_value = iter([EventList([Event(**event_data[-1])])])
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.get_events", return_value=return_value)
+
+@pytest.fixture
+def mock_get_no_events(mocker):
+    return_value = iter([EventList([])])
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.get_events", return_value=return_value)
+
+@pytest.fixture
+def mock_set_event_state(mocker):
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.set_event_state", return_value=None)
+
+@pytest.fixture
+def mock_write_events_to_lakehouse_tables(mocker):
+    return mocker.patch("cdf_fabric_replicator.event.EventsReplicator.write_events_to_lakehouse_tables", return_value=None)
+
+def snake_to_camel(snake_str):
```

Comment:
Should probably add type hints to method.

Reply:
Added type hints to both methods.